pandas/tests/test_index.py

```diff
@@ -2680,6 +2688,14 @@ def test_get_indexer(self):
         with self.assertRaisesRegexp(ValueError, 'different freq'):
             idx.asfreq('D').get_indexer(idx)
 
+    def test_repeat(self):
+        # GH10183
+        idx = pd.period_range('2000-01-01', periods=3, freq='D')
+        res = idx.repeat(3)
```

Comment:
move this as well to ``Base`` (you will have to model like other functions, e.g. iterate on ``self.indicies``